### ISSUE TITLE
[Top Bar] Missing focus ring for navigation button

### ISSUE THREAD

**Issue by lgriffee** [Shopify/polaris-react-archive#4663]
## Issue summary

Currently there appears to be no focus ring associated with the navigation button on the TopBar (`navigationButtonMarkup` in `TopBar.tsx` / `.NavigationIcon` in `TopBar.scss`).

## Expected behavior

I would expect there to be a focus ring like all the other elements in the TopBar.

## Actual behavior

[Video example](https://screenshot.click/17-05-k85ho-4lohf.mp4)
When I tab through the elements in the TopBar and the navigation button is focused it has nothing visual to indicate that it is focused.

## Steps to reproduce the problem

1. `dev s` to open Storybook in the polaris-react repo 
2. Go to the sidebar > Top bar > Top bar with all its elements
3. Tab through the elements in the TopBar

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 7.3.1
- Browser: Chrome (Version 95.0.4638.69), Safari (Version 15.1 [17612.2.9.1.20]), Firefox (Version 91.0.2)
- Device: MacBook Pro (16-inch, 2019)
- Operating System: macOS Monterey
